lib/redis.rb

```diff
@@ -2130,13 +2130,14 @@ def unwatch
 
   def pipelined
     synchronize do |client|
+      pipeline = Pipeline.new
       begin
-        original, @client = @client, Pipeline.new
+        original, @client = @client, pipeline
         yield(self)
-        original.call_pipeline(@client)
       ensure
         @client = original
       end
+      original.call_pipeline(pipeline)
```

Comment:
@byroot would `@client.call_pipeline(pipeline)` be clearer here (as `original` is defined in the `begin`)?